### PR TITLE
Replace if-assert statements with require

### PR DESCRIPTION
**Pull request by reinkrul**
Replaces if statements with an assertion and return statement with a simple require, which is functionally the same but shorter.

Just hit the like-button.

### CODE REVIEW

File: auth/services/oauth/oauth_test.go [hunk 1]
```diff
@@ -149,10 +146,9 @@ func TestAuth_CreateAccessToken(t *testing.T) {
 		signToken(tokenCtx)
 
 		response, err := ctx.oauthService.CreateAccessToken(services.CreateAccessTokenRequest{RawJwtBearerToken: tokenCtx.rawJwtBearerToken})
+
 		assert.Nil(t, response)
-		if assert.NotNil(t, err) {
-			assert.Contains(t, err.Error(), "JWT validity too long")
-		}
+		require.ErrorContains(t, err, "JWT validity too long")
```
Comment (gerardsn) [inline auth/services/oauth/oauth_test.go:151]: assert

**Review by gerardsn — CHANGES_REQUESTED** [review]
A couple of points
- [x] changing `.ldjson` context files is dangerous and should probably not be in this PR
- [ ] a couple of `require` statements too many
- [x] search for `if !assert` still finds a couple of more cases

File: network/dag/verifier_test.go [hunk 2]
```diff
@@ -128,9 +129,7 @@ func TestTransactionSignatureVerifier(t *testing.T) {
 		keyResolver.EXPECT().ResolvePublicKey(gomock.Any(), gomock.Any()).Return(nil, errors.New("failed"))
 
 		err := NewTransactionSignatureVerifier(keyResolver)(nil, d)
-		if !assert.Error(t, err) {
-			return
-		}
+		require.Error(t, err)
 
 		assert.Contains(t, err.Error(), "unable to verify transaction signature, can't resolve key by TX ref")
 		assert.Contains(t, err.Error(), "failed")
```
[the remark this answers is not in the document]
Comment (gerardsn) [reply network/dag/verifier_test.go:135]: actually is part of this PR. Drop `require.Error` and make the other `assert.ErrorContains`

File: core/echo_errors.go [hunk 3]
```diff
@@ -108,6 +108,7 @@ func PreconditionFailedError(errStr string, args ...interface{}) error {
 
 // HTTPStatusCodeError defines an interface for HTTP errors that includes a HTTP statuscode
 type HTTPStatusCodeError interface {
+	error
```
[the remark this answers is not in the document]
Comment (reinkrul) [reply core/echo_errors.go:111]: I needed it to be able to use ErrorContains in some other test

[the remark this answers is not in the document]
Comment (gerardsn) [reply core/echo_errors.go:111]: I missed the err in the implementation

**Review by gerardsn — COMMENTED** [review]
It looks like most `require.Error` statements only serve to prevent nil dereferencing in error validations using formats like `assert.Equal(t,"some error string", err.Error())`. Error validation in this format is just wrong, but it gets the job done. IMO we have (almost) no valid use-cases for `require.Error` or `require.ErrorContains`

**Comment by reinkrul** [comment]
> It looks like most `require.Error` statements only serve to prevent nil dereferencing in error validations using formats like `assert.Equal(t,"some error string", err.Error())`. Error validation in this format is just wrong, but it gets the job done. IMO we have (almost) no valid use-cases for `require.Error` or `require.ErrorContains`

Because original scope was to eliminate the if-statements but hey, we came so far so let's go the last mile. I removed the unnecessary `Error()` statements when there's also an assert on its contents.

**Pull request merged**